src/Analyzers/Core/Analyzers/UseCollectionInitializer/AbstractUseCollectionInitializerAnalyzer.cs

```diff
@@ -107,86 +149,157 @@ protected override void AddMatches(ArrayBuilder<Match<TStatementSyntax>> matches
                     continue;
                 }
 
-                if (extractedChild is TExpressionStatementSyntax expressionStatement)
+                if (extractedChild is TExpressionStatementSyntax expressionStatement &&
+                    TryProcessExpressionStatement(expressionStatement))
```

Comment:
this large if/else was extracted into separate nested helper local functions.  

One for `x.Add(y)`, one for `foreach (var y in z) x.Add(y)` and one for `if (z) x.Add(y)`